src/jit/CMakeLists.txt

```diff
@@ -126,6 +126,7 @@ if (WIN32)
     hashbv.h
     host.h
     hostallocator.h
+    hwintrinsichelperlistxarch.h
```

Comment:
I do not think a new list is necessary. Just use the original table and change `HW_Flag_NoCodeGen` intrinsics to return `nullptr` is enough.

Reply:
> I do not think a new list is necessary. Just use the original table and change HW_Flag_NoCodeGen intrinsics to return nullptr is enough.

Compiler decides which compilation path to chose based on return value from `NamedIntrinsic Compiler::lookupHWIntrinsic(const char* methodName, InstructionSet isa)` call which should be NI_Illegal. There is no point in going through HW intrinsic codegen at all, as managed implementation of intrinsic can be treated as normal function call.

Reply:
> Compiler decides which compilation path to chose based on return value from NamedIntrinsic Compiler::lookupHWIntrinsic(const char* methodName, InstructionSet isa) call which should be NI_Illegal. 

If you want to cut off the intrinsic-recognition/importation earlier, I think just removing the definition of the managed-implemented helper from this table is okay.

If you want to explicitly treat these helpers in JIT, I prefer to return `nullptr` from the importer. 
Another table is unnecessary.

Reply:
I agree that another table is unnecessary.

Reply:
> If you want to cut off the intrinsic-recognition/importation earlier, I think just removing the definition of the managed-implemented helper from this table is okay.

The point is that the way it was done at the beginning - removing definition from table - was deemed OK but the desire expressed both by @tannergooding and @CarolEidt was to have some reasonable way to control the correctness of this path via assertion.

see discussion here https://github.com/dotnet/coreclr/pull/16797#issuecomment-371172663  and here https://github.com/dotnet/coreclr/pull/16797#issuecomment-371294081

There is still one more way to do it without introducing additional complexity and new table and I will create separate branch to allow CR of that option

Reply:
> but the desire expressed both by @tannergooding and @CarolEidt was to have some reasonable way to control the correctness of this path via assertion.

A simple flag `HW_Flag_ManagedImplementation` (or something with a better name) should be completely sufficient for this.

Reply:
> The point is that the way it was done at the beginning - removing definition from table - was deemed OK but the desire expressed both by @tannergooding and @CarolEidt was to have some reasonable way to control the correctness of this path via assertion.

That is another story. As I said, there are two solutions:
1. Managed-implemented helpers cannot reach the importer of HW intrinsics if the definition is removed from the table.

2. Checking the flag of `HW_Flag_NoCodeGen` (or another new name) and returning `nullptr` can solve the "control the correctness of this path via assertion." that @tannergooding and @CarolEidt mentioned.

Reply:
BTW, I prefer that "Checking the flag of HW_Flag_NoCodeGen (or another new name) and returning nullptr", which let the managed-implemented helpers can leverage `NotSupported` checking (e.g., invalid type-argument for generic intrinsic) and managed code does not need to care about throwing the exception.

Reply:
You used the first solution (removing the definition) in https://github.com/dotnet/coreclr/pull/16817. Please unify.

Reply:
That is just obvious and was indicated several times that #16817 has to be merged first to work on final solution here which will unfiy both and this is what I am doing here - the code in current version unified all implementations and the code in new version  will unify them as well.

Reply:
> That is just obvious and was indicated several times

> the code in current version unified all implementations and the code in new version will unify them as well.

Sounds good. 
I was in traveling recently, so that I missed some conversation contexts. Sorry to duplicate the discussion.